dali/base/dafdesc.cpp

```diff
@@ -3699,12 +3699,20 @@ static void doInitializeStorageGroups(bool createPlanesFromGroups)
             if (cur.groupType == grp_thor)
                 cur.createStoragePlane(storage, 1);
         }
+    }
 
-        //Uncomment the following to trace the values that been generated
-        //printYAML(storage);
+    // Add groups defined in environment.xml
+    Owned<IPropertyTree> env = getHPCCEnvironment();
+    Owned<IPropertyTreeIterator> storagePlanes = env->getElements("Software/Globals/StoragePlaneList/StoragePlane[@name='*']");
```

Comment:
minor:I would use "Storage" (or "StoragePlanes" rather than "StoragePlaneList" 

Also, no need to have a qualifier on the name:

```
Owned<IPropertyTreeIterator> storagePlanes = env->getElements("Software/Globals/StoragePlanes/StoragePlane");
```
I am not sure about the benefits of accessing via the environment, or by copying the data to the component configuration file and accessing from there.
@jakesmith opinions?

Reply:
The standard approach is to generate into the component config file (in this case daliconf.xml).
We do this already for other components (dfuserver, esp.. ), see e.g. initfiles/componentfiles/configxml/dfuserver.xsl. At the moment it does this for RemoteStorage only in those places.

<planes> should be generated in the same way into those configs too, and Dali (which doesn't need RemoteStorage) should generate <planes> only.

One caveat to that is that the above code only creates planes for existing groups in Dali (that correspond to engine groups in BM) if "planes" doesn't exist, but it will do after generating it into the config..

I suggest changing the createStoragePlane calls, so that in this context it adds a flag to the planes being added, to indicate they are based on groups (e.g. @fromGroup=true), and then change the start of this process to:
```
    if (createPlanesFromGroups)
    {
        // delete all [@fromGroup=true] planes 
        
        // existing code that constructs groups and adds planes for groups


...
     }
```

NB: this routine can be recalled if/when environment.xml changes.

Reply:
Copied storage/planes definitions into component config files which allowed a group to be created for my storage plane.

I changed createStoragePlane to add the flag fromGroup to the planes created from it. Then at the beginning of the process changed it to remove all storage planes previously created by createStoragePlane by removing any with fromGroup=true. I also had to add a check at the beginning of createStoragePlanes for a storage plane that is already defined (this would be coming from the component config file) otherwise it would create a duplicate planes.